deploy/docs/Best_Practices.md

```diff
@@ -511,6 +530,64 @@ You can find more information on the `grep` filter plugin in the
 Refer to our [documentation](v1_conf_examples.md) for other examples of how you can
 customize the fluentd pipeline.
 
+## Modifying logs in Fluentd
+
+You can redact log messages in order to e.g. prevent sending sensitive data like passwords.
+
+In order to do that [record_transformer filter][record_transformer] can be used.
+Please consider the following configuration:
+
+```yaml
+fluentd:
+  logs:
+    containers:
+      extraFilterPluginConf: |-
+          <filter containers.var.log.pods.*_*_*.test-container.*.log containers.var.log.containers.*_*_test-container-*.log>
+            @type record_transformer
+            enable_ruby
+            <record>
+              # Replace `Password: <pass>` with `Password: ***`
+              log ${record["log"].gsub(/Password: (.*)/, 'Password: ***')}
+              # Replace kubernetes['namespace_name'] with 'REDACTED'
+              # if namespace_name exists in record['kubernetes'], then change value of it and then return modified record['kubernetes'] to assign as kubernetes
+              kubernetes = ${if record['kubernetes'].has_key?('namespace_name'); record['kubernetes']['namespace_name'] = "REDACTED"; end; record['kubernetes']}
+            </record>
+          </filter>
+```
+
+It is going to replace all `Password: <pass>` occurence in logs with `Password: ***` and replace namespace to `REDACTED`
+for [all containers](#difference-between-varlogpods-and-varlogcontainers) named `test-container`.
+
+An example log before entering the `extraFilterPluginConf` section is presented below:
+
+```json
+{
+  "stream": "stdout",
+  "logtag": "F",
+  "log": "log... ",
+  "docker": {
+    "container_id": "a1acfd70-c8d1-456b-95dd-515f1256906f"
+  },
+  "kubernetes": {
+    "container_name": "example-container-multiline-logs-long-lines",
+    "namespace_name": "multiline-logs-generator",
```

Comment:
I know that this is just example but it will better to have this example more related to above configuration so I'd prefer container name set to `test-container`.

Reply:
updated